packages/playground/website/src/github/preview-pr/form.tsx

```diff
@@ -0,0 +1,233 @@
+import { setActiveSiteError } from '../../lib/state/redux/slice-ui';
+import { useDispatch } from 'react-redux';
+import { PlaygroundDispatch } from '../../lib/state/redux/store';
+import ModalButtons from '../../components/modal/modal-buttons';
+
+interface PreviewPRFormProps {
+	onImported: () => void;
+	onClose: () => void;
+	target: 'wordpress' | 'gutenberg';
+}
+
+const urlParams = new URLSearchParams(window.location.search);
+
+export const targetParams = {
+	'wordpress': {
+		repo: 'wordpress-develop',
+		workflow: 'Test%20Build%20Processes',
+		artifact: 'wordpress-build-'
+	},
+	'gutenberg': {
+		repo: 'gutenberg',
+		workflow: 'Build%20Gutenberg%20Plugin%20Zip',
+		artifact: 'gutenberg-plugin'
+	}
+}
+
+export default function PreviewPRForm({
+	onImported,
+	onClose,
+	target = 'wordpress'
+}: PreviewPRFormProps) {
+	const dispatch: PlaygroundDispatch = useDispatch();
+	const [value, setValue] = useState<string>('');
+	const [submitting, setSubmitting] = useState<boolean>(false);
+	const [errorMsg, setError] = useState<string>('');
+
+	useEffect(() => {
+		const query = new URLSearchParams(window.location.search);
+		if (query.has('core-pr')) {
+			const prNumber = query.get('core-pr');
+			prNumber && setValue(prNumber);
+		}
+	}, []);
+
+	async function handleSubmit(e: React.MouseEvent<HTMLButtonElement>) {
+		e.preventDefault();
+
+		if (!value) {
+			return;
+		}
+
+		await previewPr(value);
+	}
+
+	function renderRetryIn(retryIn: number) {
+		setError(`Waiting for GitHub to finish building PR ${value}. This might take 15 minutes or more! Retrying in ${
+			retryIn / 1000
+		}...`);
+	}
+
+	async function previewPr(prValue: string) {
+		let cleanupRetry = () => {};
+		if (cleanupRetry) {
+			cleanupRetry();
+		}
+
+		let prNumber: string = prValue;
+		setSubmitting(true);
+
+		// Extract number from a GitHub URL
+		if (
+			prNumber
+				.toLowerCase()
+				.includes('github.com/wordpress/wordpress-develop/pull') ||
+			prNumber
+				.toLowerCase()
+				.includes('github.com/wordpress/gutenberg/pull')
+		) {
+			// @ts-ignore
+			prNumber = prNumber.match(/\/pull\/(\d+)/)[1];
+		}
+
+		// Verify that the PR exists and that GitHub CI finished building it
+		// @ts-ignore
+		const zipArtifactUrl = `https://playground.wordpress.net/plugin-proxy.php?org=WordPress&repo=${targetParams[target].repo}&workflow=${targetParams[target].workflow}&artifact=${targetParams[target].artifact}${target === 'wordpress' ? prNumber : ''}&pr=${prNumber}`;
+		// Send the HEAD request to zipArtifactUrl to confirm the PR and the artifact both exist
+		const response = await fetch(zipArtifactUrl + '&verify_only=true');
+		if (response.status !== 200) {
+			let error = 'invalid_pr_number';
+			try {
+				const json = await response.json();
+				if (json.error) {
+					error = json.error;
+				}
+			} catch (e) {
+				logger.error(e);
+				// TODO: Is it right error here?
+				dispatch(setActiveSiteError('site-boot-failed'));
```

Comment:
This error is used when Playground fails to boot.
Why do we need to dispatch a site error here? Would it work to just set an error in the form?

From what I can see this will happen when the plugin proxy returns a response that isn't a valid JSON. 
Is there an error in the response that we could use in this case? If no, we could do something generic like _We couldn't verify your PR, please try again._.